brainscore/metrics/nsd_algonauts_correlation.py

```diff
@@ -0,0 +1,129 @@
+
+import numpy as np
+from sklearn.linear_model import LinearRegression
+from brainscore.metrics import Score
+from sklearn.preprocessing import StandardScaler
+from sklearn.decomposition import PCA
+from tqdm import tqdm
+from scipy.stats import pearsonr as corr
+
+class RegressedCorrelationAlgonauts:
+    ''' 
+    This class performs regression-based correlation analysis on brain imaging data.
+    It takes source and target data, preprocesses the data, applies regression,
+    and calculates correlations.
+    The train and test split, as well as the ceiling, are calculated 
+    accordingly to algonauts 2023 challenge.
+    '''
+    def __init__(self, regression, correlation):
+        regression = regression or linear_regression
+
+        self.regression = regression
+        self.correlation = correlation
+    
+    def __call__(self, source, target):
+        
+            # =============================================================================
+            # Extracting the features from the ANN
+            # =============================================================================
+            print('initial source shape: ', source.shape)
+            print('initial target shape: ', target.shape)
+
+            print('source.stimulus_id: ', source.stimulus_id.data)
+            print('target.stimulus_id: ', target.stimulus_id.data)
+        
+            # divide into train and test
+            train_idx = np.where(target.algonauts_train_test == 'train')[0]
+            test_idx = np.where(target.algonauts_train_test == 'test')[0]
```

Comment:
where does `algonauts_train_test` come from? This metric is confusing to me

Reply:
It's created during the packaging of the stimulus set, to give info on whether each stimulus is part of train and test during the algonauts challenge. It's only used ro reprodouce algonauts results, and will be ignored by other metrics.